WordPress/Classes/ViewRelated/Pages/PageListViewController.m

```diff
@@ -0,0 +1,531 @@
+    if (!objectID) {
+        return nil;
+    }
+
+    NSError *error = nil;
+    Blog *restoredBlog = (Blog *)[context existingObjectWithID:objectID error:&error];
+    if (error || !restoredBlog) {
+        return nil;
+    }
+
+    return [self controllerWithBlog:restoredBlog];
+}
+
+- (void)encodeRestorableStateWithCoder:(NSCoder *)coder
+{
+    [coder encodeObject:[[self.blog.objectID URIRepresentation] absoluteString] forKey:PagesViewControllerRestorationKey];
+    [super encodeRestorableStateWithCoder:coder];
+}
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
+{
+    self.postListViewController = segue.destinationViewController;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    self.title = NSLocalizedString(@"Pages", @"Tile of the screen showing the list of pages for a blog.");
+}
+
+
+#pragma mark - Configuration
+
+- (void)configureCellsForLayout
+{
+    self.cellForLayout = (PageListTableViewCell *)[[[NSBundle mainBundle] loadNibNamed:PageCellNibName owner:nil options:nil] firstObject];
+}
+
+- (void)configureTableView
+{
+    self.tableView.accessibilityIdentifier = @"PagesTable";
+    self.tableView.isAccessibilityElement = YES;
+    self.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
+
+    // Register the cells
+    UINib *pageCellNib = [UINib nibWithNibName:PageCellNibName bundle:[NSBundle mainBundle]];
+    [self.tableView registerNib:pageCellNib forCellReuseIdentifier:PageCellIdentifier];
+
+    UINib *restorePageCellNib = [UINib nibWithNibName:RestorePageCellNibName bundle:[NSBundle mainBundle]];
+    [self.tableView registerNib:restorePageCellNib forCellReuseIdentifier:RestorePageCellIdentifier];
+}
+
+- (NSString *)noResultsTitleText
+{
+    if (self.syncHelper.isSyncing) {
+        return NSLocalizedString(@"Fetching pages...", @"A brief prompt shown when the reader is empty, letting the user know the app is currently fetching new pages.");
+    }
+    PostListFilter *filter = [self currentPostListFilter];
+    NSString *title;
+    switch (filter.filterType) {
+        case PostListStatusFilterDraft:
+            title = NSLocalizedString(@"You don't have any drafts.", @"Displayed when the user views drafts in the pages list and there are no posts");
+            break;
+        case PostListStatusFilterScheduled:
+            title = NSLocalizedString(@"You don't have any scheduled pages.", @"Displayed when the user views scheduled pages in the pages list and there are no pages");
+            break;
+        case PostListStatusFilterTrashed:
+            title = NSLocalizedString(@"You don't have any pages in your trash folder.", @"Displayed when the user views trashed in the pages list and there are no pages");
+            break;
+        default:
+            title = NSLocalizedString(@"You haven't published any pages yet.", @"Displayed when the user views published pages in the pages list and there are no pages");
+            break;
+    }
```

Comment:
Plus, even if `filter == nil`, nothing would break. Worst case scenario, you can always handle defaults.

```
PostListStatusFilter filter = filter ? filter.filterType : PostListStatusFilterDefault
```